ostd/src/arch/x86/irq.rs

```diff
@@ -58,6 +58,18 @@ pub(crate) fn disable_local() {
     x86_64::instructions::interrupts::disable();
 }
 
+pub(crate) fn enable_local_if(cond: bool) {
+    if cond {
+        enable_local();
+    }
+}
+
+pub(crate) fn disable_local_if(cond: bool) {
```

Comment:
These are mostly helper functions, so I think `trap/mod.rs` (e.g., put them just below `trap_handler`) is a better place for them. At least for now, no one else is using them.